inlanecruising_plugin/src/inlanecruising_plugin.cpp

```diff
@@ -228,108 +229,99 @@ std::vector<cav_msgs::TrajectoryPlanPoint> InLaneCruisingPlugin::compose_traject
   log::printDebugPerLine(time_bound_points, &log::pointSpeedPairToStream);
 
   ROS_DEBUG("Got basic points ");
-  std::vector<DiscreteCurve> sub_curves = compute_sub_curves(time_bound_points);
-
-  ROS_DEBUG_STREAM("Got sub_curves " << sub_curves.size());
 
   std::vector<double> final_yaw_values;
   std::vector<double> final_actual_speeds;
   std::vector<lanelet::BasicPoint2d> all_sampling_points;
 
-  for (const auto& discreet_curve : sub_curves)
-  {
-    ROS_DEBUG("SubCurve");
-
-    std::vector<double> speed_limits;
-    std::vector<lanelet::BasicPoint2d> curve_points;
-    splitPointSpeedPairs(discreet_curve.points, &curve_points, &speed_limits);
-
-    std::unique_ptr<smoothing::SplineI> fit_curve = compute_fit(curve_points); // Compute splines based on curve points
-
-    if (!fit_curve)
-    {  // TODO how better to handle this case
-      for (size_t i = 0; i < discreet_curve.points.size() - 1; i++)
-      {
-        Eigen::Isometry2d point_in_map =
-            curvePointInMapTF(discreet_curve.frame, discreet_curve.points[i].point, final_yaw_values.back());
-        all_sampling_points.push_back(point_in_map.translation());
-        final_yaw_values.push_back(final_yaw_values.back());
-        final_actual_speeds.push_back(final_actual_speeds.back());
-      }
-      continue;
+  std::vector<double> speed_limits;
+  std::vector<lanelet::BasicPoint2d> curve_points;
+  splitPointSpeedPairs(time_bound_points, &curve_points, &speed_limits);
+  
+  std::unique_ptr<smoothing::SplineI> fit_curve = compute_fit(curve_points); // Compute splines based on curve points
+  if (!fit_curve)
+  {  // TODO how better to handle this case
+    ROS_ERROR_STREAM("Could not fit!");
+    for (size_t i = 0; i < curve_points.size() - 1; i++)
+    {
+      all_sampling_points.push_back(curve_points[i]);
+      final_yaw_values.push_back(final_yaw_values.back());
```

Comment:
I think this yields undefined behavior per the C++ standard if the vector is empty, you're effectively referencing an uninitialized element.